File: stara/2006/python/zad6_a.py
# wczytanie slow
with open('../dane/dane.txt') as f:
    data = []
    for line in f.readlines():
        data.append(line[:-1])


# slownik zawierajacy wszystkie slowa i ich ilosc
words = {}
for word in data:
    # jezeli jest juz zapisane dodaj jeden do licznika, inczej stworz z licznikiem 1
    if word in words:
        words[word] += 1
    else:
        words[word] = 1

# licznik powtarzajacyh sie slow
repeats = 0
# najczestsze slowo
most_common = data[0]

for word in words:
    # jezeli jest licznik wskazuje wiecej niz jeden zlicz
    if words[word] > 1:
        repeats += 1
    # jezeli pobilo rekord zapisz
    if words[most_common] < words[word]:
        most_common = word


# wyswietlenie odpowiedzi
answer = f'6 a) Ilosc slow wystepujacych wiecej niz raz: {repeats}; Slowo o najwiekszej liczbie wystapien: {most_common}; Wystapil tyle razy: {words[most_common]}'
print(answer)

File: stara/2006/c++/zad6_a.cpp
#include <iostream>
#include <string>
#include <fstream>
#include <vector>
using namespace std;

// funkcja sprawdzajaca czy w danym vectorze dane slowo wystepuje wiecej niz jeden raz
bool isMoreThanOne(string word, vector<string> vec, vector<string> banned)
{
    for (int i = 0; i < banned.size(); i++)
    {
        if (word == banned[i])
        {
            return false;
        }
    }

    int counter = 0;

    for (int i = 0; i < vec.size(); i++)
    {
        if (word == vec[i])
        {
            counter++;
        }
    }

    if (counter > 1)
    {
        return true;
    }
    else
    {
        return false;
    }
}

string zad6_a()
{
    string line;
    fstream file("../dane/dane.txt");
    vector<string> content;

    // wczytywanie elementow z pliku
    if (file.is_open())
    {
        while (getline(file, line))
        {
            content.push_back(line);
        }
    }
    file.close();

    int howManyMoreThanOne = 0;
    vector<string> banned;

    for (int i = 0; i < content.size(); i++)
    {
        if (isMoreThanOne(content[i], content, banned))
        {
            howManyMoreThanOne++;
            banned.push_back(content[i]);
        }
    }

    // sprawdzanie, ktore slowo wystepuje najwiecej razy w pliku
    int max = 0;
    string currentMax = "h";

    for (int i = 0; i < content.size(); i++)
    {
        int temp = 0;
        
        for (int j = 0; j < content.size(); j++)
        {

            if (content[i] == content[j])
            {
                temp++;
            }
        }

        if (temp > max)
        {
            max = temp;
            currentMax = content[i];
        }
    }

    // cout << currentMax;

    string answer = "6 a) Liczba slow wystepujacych w pliku wiecej niz jeden raz: " + to_string(howManyMoreThanOne);

    return answer;
}
